### PR TITLE
Update Rust toolchain to 1.63.0

### PR DESCRIPTION
**Pull request by jcgruenhage**
#### Testing the changes
- I tested the changes in this PR: **briefly**

[ci skip]

`cargo-crev` requires 1.62 at least, so I thought I'd gave updating Rust a go, and it worked out just fine tbh. The new `cargo-crev` release compiled fine on this toolchain and works properly.

#### Local build testing
- I built this PR natively for
  - `x86_64`, `glibc` and `musl`
  - `i686`, `glibc`
- I crossbuilt this PR for
  - `aarch64`, `musl`
  - `ppc`, `glibc`
  - `ppc64`, `glibc`
  - `ppc64le`, `musl`

### CODE REVIEW

**Comment by jcgruenhage** [comment]
@q66 Could You generate the various ppc distfiles for rust and cargo for the new update?

**Comment by jcgruenhage** [comment]
@q66 Actually, how would I crossbuild the bootstrap distfiles for the PPC architectures? Just doing `./xbps-src -a ppc64 pkg -o bindist rust` [doesn't seem to do the trick](https://pastebin.com/DSnDn0vt)

**Comment by classabbyamp** [comment]
I think you need to add the ppc repos to `etc/xbps.d/repos-remote*.conf`

**Comment by jcgruenhage** [comment]
I'll try that tomorrow then, thanks for the pointer. I assume the builds are reproducible enough that whatever I generate locally will match the package generated by the void ppc build system?.

**Comment by Zapeth** [comment]
I noticed that the `rust.description` field in the config.toml hasn't been set yet ([reference](https://github.com/rust-lang/rust/blob/4916e2b9e6ef8cee6f9c6abb75bd01ba9dc07e5c/config.toml.example#L521-L524)), could you maybe also include that here?

**Comment by jcgruenhage** [comment]
> I noticed that the `rust.description` field in the config.toml hasn't been set yet ([reference](https://github.com/rust-lang/rust/blob/4916e2b9e6ef8cee6f9c6abb75bd01ba9dc07e5c/config.toml.example#L521-L524)), could you maybe also include that here?

Added this now, thanks for the pointer.

**Comment by jcgruenhage** [comment]
Sorry for the hold-up, I'm currently building the bootstrap packages and really am in need for a faster build machine, lol, waiting hours per architecture isn't exactly great.

File: srcpkgs/rust/template [hunk 1]
```diff
@@ -127,6 +127,7 @@ do_configure() {
 		incremental = false
 		parallel-compiler = false
 		channel = "stable"
+		description = "Void"
```
Comment (jcgruenhage) [inline srcpkgs/rust/template:130]: Causes this suffix in the compiler version output:
```
$ rustc --version
rustc 1.63.0 (Void)
```

File: srcpkgs/rust/patches/0010-Fix-dynamic-linkage-of-musl-libc-for-the-libc-crate.patch [hunk 2]
```diff
@@ -1,33 +1,31 @@
...
+@@ -337,11 +337,11 @@ cfg_if! {
          #[link(name = "c", cfg(not(target_feature = "crt-static")))]
          extern {}
      } else if #[cfg(target_env = "musl")] {
 +        #[link(name = "c")]
 +        extern {}
          #[cfg_attr(feature = "rustc-dep-of-std",
 -                   link(name = "c", kind = "static", modifiers = "-bundle",
-+                   link(name = "gcc", kind = "static-nobundle",
++                   link(name = "gcc", kind = "static", modifiers = "-bundle",
```
Comment (jcgruenhage) [inline srcpkgs/rust/patches/0010-Fix-dynamic-linkage-of-musl-libc-for-the-libc-crate.patch:22]: `kind = "static-nobundle"` was removed. `static` defaults to the modifier `+bundle`, and setting the modifier `-bundle` behaves like ``static-nobundle` behaved before.

File: srcpkgs/rust/template (hunk 1 above)
Comment (classabbyamp) [reply srcpkgs/rust/template:130]: hm maybe `Void Linux`? (a bit of a nitpick)

Comment (jcgruenhage) [reply srcpkgs/rust/template:130]: We can do that for the next version, but I'd prefer not rebuilding rust another 7 times to get the new distfiles into place with that change included :D

**Comment by jcgruenhage** [comment]
The self-hosted distfiles for ppc64, ppc64le-musl and ppc have been uploaded, which probably means that this is now relatively good-to-go?

File: srcpkgs/rust/template (hunk 1 above)
Comment (classabbyamp) [reply srcpkgs/rust/template:130]: it's fine for the bootstrap distfiles to not have that minor difference I think, I'll make the change when merging.

**Comment by classabbyamp** [comment]
will merge when firefox issues on the builders get fixed

**Pull request merged**